src/string.cr

```diff
@@ -1525,9 +1527,18 @@ class String
           byte = to_unsafe[i]
           if byte < 0x80
             char = byte.unsafe_chr
-            replaced_char = upcase_next ? char.upcase : char.downcase
+            replaced_char = if upcase_next
+                              upcase_next = false
+                              char.upcase
+                            elsif underscore_to_space && '_' == char
+                              upcase_next = true
+                              ' '
+                            else
+                              upcase_next = char.ascii_whitespace?
+                              char.downcase
+                            end
```

Comment:
I'd prefer if `upcase_next` and `replaced_char` are multi-assigned, instead of having each clause assign to `upcase_next` separately